h5py/tests/test_file.py

```diff
@@ -1024,6 +1024,24 @@ def test_ros3(self):
             assert 'mydataset' in f.keys()
             assert f["mydataset"].shape == (100,)
 
+    def test_ros3_s3_fails(self):
+        """ROS3 exceptions for s3:// location"""
+        with pytest.raises(ValueError,
+                           match='AWS region required for s3:// location'):
+            File('s3://fakebucket/fakekey', 'r', driver='ros3')
+
+        with pytest.raises(ValueError,
+                           match=r'^foo://wrong/scheme: S3 location must begin with'):
+            File('foo://wrong/scheme', 'r', driver='ros3')
+
+    def test_ros3_s3uri(self):
```

Comment:
Can we put this in its own file or make it out-in?  I'm super happy to pick up a network dependency in the test suite.

I think we also need to add a skip on both of these if h5py is not built with ros.

Reply:
Don't know what "make it out-in" means.

I can certainly put all ROS3 tests in a single file, I just added these two to an already existing class.

The entire test class will now be skipped if the ros3 driver is not available:
https://github.com/h5py/h5py/blob/5006c597fce7054b516c93a91cc69ff8a5808cb1/h5py/tests/test_file.py#L1014-L1017

Reply:
@tacaswell I moved the ros3 tests into a separate file.

Reply:
meant "opt-in" sorry, typing failure 🐑 .

Reply:
The ros3 tests are now in their own file: [test_ros3.py](https://github.com/h5py/h5py/blob/c70487c8d91ea23969a0b1b6d8171bcd53cc19d0/h5py/tests/test_ros3.py). The tests in this file will be skipped for HDF5 prior to 1.10.6 or if HDF5 was not built with the ros3 driver. Is this now acceptable?

Reply:
I think we may still want to add a "--network" or "--no-network" flag.  I know that some of the linux distros run their build system in contexts where there is no network access so it is good if we could give them a way to disable the test that actually reads from s3 without having to patch the tests.

Reply:
Added `--no-network` CLI option to pytest. Created a new `pytest.mark.nonetwork" marker and added it to the ros3 tests.

Reply:
@tacaswell Are the current changes acceptable to resolve this conversation? When I run `pytest -rs --no-network h5py` the ros3 tests are all skipped:
```
SKIPPED [3] h5py/tests/test_ros3.py: No Internet
```